classes/controller/FrontController.php

```diff
@@ -391,21 +391,7 @@ public function init()
                 $cart->update();
             }
             /* Select an address if not set */
-            if (isset($cart) && (!isset($cart->id_address_delivery) || $cart->id_address_delivery == 0 ||
```

Comment:
I'm aware this is violent...  do you guys think it could be considered as a BC ? Imho this code was very problematic and was partly the cause of the bug this PR is fixing. 

This is old code from 2011, I just hope we don't have features relying on delivery and invoice addresses being updated in the FrontController 😓

Reply:
> This is old code from 2011, I just hope we don't have features relying on cart's delivery and invoice addresses being updated in FrontController::init() 😓

I expect that we have some 😓  😭  although the best people to tell us this would be people relying on such mechanisms to build their FO modules.

Would it be possible to get some insights from @v4lux @Quetzacoalt91 @PululuK @kpodemski @jf-viguier @prestarocket @prestamodule ? Do you think this little bit of logic "if there is a cart but no address, I pick up the first address I can use" is used and necessary ?

Reply:
If (as I am afraid) this bit of logic must be kept, maybe we can disable it for your usecase.

I'm thinking of introducing a new property `$automaticallyAllocateAnAddress = true` inside FrontController
and your usecase we must modify it to `false` in `init()` to avoid the logic being triggered

But I agree with you this little "magic" trick seems to be a potential cause for lot of weird behaviors 😞

Reply:
Yeah be careful with that, I can easily imagine with this code, we may consider that a cart will always have a delivery & invoice address a long as at least one is defined in the account.

For instance, this could be used to estimate early (= before the checkout) how much the shipping costs.

Reply:
@Quetzacoalt91 So, in the end, I only disabled the forceful assignment of the invoice address id (not delivery), and only for OrderController.